### ISSUE TITLE
Systray icon double (or single) click not working (show / hide window) (Ubuntu MATE 16.04) 

### ISSUE THREAD

**Issue by mizuapo** [yakyak/yakyak#338]
When i double clicked the systray icon then nothing don't happens.
- MATE Desktop Environment 1.12.1 based ubuntu 16.04  (gnome 2)

**Comment by rippetoej** [comment]
Same happens in Windows 10 v1607.  However, if I right click on the tray icon and then left click on "Toggle minimize to tray" the window exhibits the show/hide functionality.

**Comment by pmsobrado** [comment]
Also, if "Close to tray" is enabled, when you close the app, it closes from tray too, when it should remain on the tray. Happens on Ubuntu and Windows.

**Comment by nerijus** [comment]
Yes, it would be nice if a single click on the tray icon opened/closed the main window.

**Comment by nerijus** [comment]
"if "Close to tray" is enabled, when you close the app, it closes from tray too, when it should remain on the tray" - the following patch helps for Linux (it is probably wise to remove "if (process.platform...)" at all to work on Windows too):
```
diff -ur app.orig/main.js app/main.js
--- app.orig/main.js    2016-09-20 10:55:40.000000000 +0300
+++ app/main.js 2016-09-21 16:57:11.160924016 +0300
@@ -236,6 +236,10 @@
       reconnectCount = 0;
       return client.disconnect();
     });
+    mainWindow.on('minimize', function(ev) {
+      ev.preventDefault()
+        mainWindow.hide();
+    });
     mainWindow.on('resize', function(ev) {
       return ipcsend('resize', mainWindow.getSize());
     });
@@ -454,7 +458,7 @@
       });
     });
     return mainWindow.on('close', function(ev) {
-      if (process.platform === 'darwin') {
+      if (process.platform !== 'win32') {
         if (mainWindow.isFullScreen()) {
           mainWindow.setFullScreen(false);
         }
```
Also it hides window to tray instead of minimizing.

**Comment by nerijus** [comment]
As for the left click, it seems to be Electron on Linux limitation - https://github.com/electron/electron/issues/6773

**Comment by pmsobrado** [comment]
Let's hope there's a fix soon :)

**Comment by nerijus** [comment]
Found it. This patch lets tray left click on Linux (could be except on Ubuntu) to activate window:
```
diff -ur app.orig/ui/views/trayicon.js app/ui/views/trayicon.js
--- app.orig/ui/views/trayicon.js       2016-09-20 10:55:40.000000000 +0300
+++ app/ui/views/trayicon.js    2016-09-22 16:11:24.829826470 +0300
@@ -44,7 +44,7 @@
   create = function() {
     tray = new Tray(trayIcons["read"]);
     tray.setToolTip('YakYak - Hangouts client');
-    return tray.on('clicked', function() {
+    return tray.on('click', function() {
       return action('showwindow');
     });
   };
```

**Comment by nerijus** [comment]
And this patch lets left click to toggle window:
```
diff -ur app.orig/ui/views/trayicon.js app/ui/views/trayicon.js
--- app.orig/ui/views/trayicon.js       2016-09-20 10:55:40.000000000 +0300
+++ app/ui/views/trayicon.js    2016-09-22 16:36:47.927195129 +0300
@@ -44,8 +44,8 @@
   create = function() {
     tray = new Tray(trayIcons["read"]);
     tray.setToolTip('YakYak - Hangouts client');
-    return tray.on('clicked', function() {
-      return action('showwindow');
+    return tray.on('click', function() {
+      return action('togglewindow');
     });
   };

```

**Comment by arnriu** [comment]
@nerijus can you work with `src/ui/views/trayicon.coffee` instead of `app/ui/views/trayicon.js` ?

**Comment by nerijus** [comment]
Similar changes, not a big deal to make them yourself :)

**Comment by arnriu** [comment]
What I mean, is that you could do it in coffee files directly since it's similar, then commit your changes, and then, when you feel it's ready, make a pull request. Because it's a pain to test this way, and it won't go master unless you pr first (which means you committed, which means you worked with coffee, see ? :p ).

**Comment by nerijus** [comment]
I never did pr before, could you please check if I did it correctly?

**Comment by arnriu** [comment]
@nerijus are you sure you commited all your changes?

**Comment by nerijus** [comment]
Not all, only 2 for now.

**Comment by arnriu** [comment]
Well, it looks good so far then, i see 2 commit in your PR, with one file change in each. If you keep committing on your fork, it will add to your PR.

**Comment by averissimo** [comment]
@mizuapo @pmsobrado @rippetoej can you check if latest beta fixes this problem?

see #451 for download link

credits to @nerijus for the fix :)

**Comment by pmsobrado** [comment]
Now it hides to tray, but double (or single) click on tray to close/open
doesn't seem to work.

Anyway, good job!



* Pablo Manuel Sobrado Gómez*
 Projects area - Devops
 * Emergya Consultoría*
 Tfno: +34 954 51 75 77
 Fax: +34 954 51 64 73
 www.emergya.es


2016-11-29 20:27 GMT+01:00 André Veríssimo <notifications@github.com>:

**Comment by nerijus** [comment]
Works here on Fedora. Could be Ubuntu/Unity problem.

**Comment by pmsobrado** [comment]
I'm on Ubuntu with KDE installed. But is not a major issue to me. I'll
check on Windows when I have the time.

Thanks!



* Pablo Manuel Sobrado Gómez*
 Projects area - Devops
 * Emergya Consultoría*
 Tfno: +34 954 51 75 77
 Fax: +34 954 51 64 73
 www.emergya.es


2016-11-30 12:51 GMT+01:00 Nerijus Baliūnas <notifications@github.com>:

**Comment by averissimo** [comment]
I've tested in Xfce and seems to work flawlesly

There are some specifics on the window manager, for instance it fails miserably in unity, but that's an unity problem.

It would be awesome if you could give feedback in windows :)

![outsxu67pu8ax](https://cloud.githubusercontent.com/assets/211358/20756572/13b1f612-b70b-11e6-8071-2caed49d5250.gif)

**Comment by pmsobrado** [comment]
I'll check on Windows as soon as I can :)



* Pablo Manuel Sobrado Gómez*
 Projects area - Devops
 * Emergya Consultoría*
 Tfno: +34 954 51 75 77
 Fax: +34 954 51 64 73
 www.emergya.es


2016-11-30 15:43 GMT+01:00 André Veríssimo <notifications@github.com>:

**Comment by rippetoej** [comment]
All tray features appear to work correctly on windows 10 v1607

- Single click on try icon toggles window 
- Close to tray works
- Pressing the minimize button on the window minimizes to task bar instead of tray (I assume this is the intended action)

Is the "Toggle minimize to tray" item in the tray suppose to activate/deactivate the minimize to tray action or is it meant as another way of toggling the window between maximized/minimized? It currently toggles the window so I am guessing it is just another method for doing so.

**Comment by nerijus** [comment]
It's another method, as single click toggle does not work on Ubuntu.

**Comment by averissimo** [comment]
included in 1.4.0 release
